components/VoteTicker/VoteTicker.tsx

```diff
@@ -0,0 +1,130 @@
+import { useState, useEffect } from "react";
+import styled from "styled-components";
+import Clock from "public/assets/clock.svg";
+import UpRightArrow from "public/assets/up-right-arrow.svg";
+import calculateTimeRemaining from "./calculateTimeRemaining";
+
+type theme = "light" | "dark";
+
+interface Props {
+  theme: theme;
+}
+
+const VoteTicker: React.FC<Props> = ({ theme }) => {
+  const { timeRemaining } = useVoteTicker();
+  return (
+    <Section theme={theme}>
+      <Wrapper theme={theme}>
+        <VoteBlock theme={theme}>
+          <ClockBG theme={theme}>
+            <Clock />
+          </ClockBG>
+          <VoteText theme={theme}>
+            Time to commit vote:
+            <span>{timeRemaining}</span>
+          </VoteText>
+        </VoteBlock>
+        <MoreDetailsBlock theme={theme}>
+          <span>More details</span>
+          <a href="https://vote.umaproject.org/" target="_blank" rel="noreferrer">
+            <UpRightArrow />
+          </a>
+        </MoreDetailsBlock>
+      </Wrapper>
+    </Section>
+  );
+};
+
+export default VoteTicker;
+
+function useVoteTicker() {
+  const [timeRemaining, setTimeRemaining] = useState("00:00");
+  // Set time remaining depending if it's the Commit or Reveal
+  // Note: the requests are all slightly differently in there final vote time. I'll use the last
+  // Vote added.
+  useEffect(() => {
+    setTimeRemaining(calculateTimeRemaining());
+
+    const timer = setInterval(() => {
```

Comment:
Doing this in a hook is going to create new intervals every time the hook gets called in a component, which might not be what you want. [This](https://usehooks-ts.com/react-hook/use-interval) could help.

Reply:
Love me some good generic hooks, I'll implement this.